Add Shape inference for CopyOp for lazy tensor core backend

rahuls-cerebras opened the pull request:
This PR adds the shape inference loging for CopyOp for lazy tensor core backend

rahuls-cerebras commented:
Gentle Reminder @ramiro050

GlebKazantaev commented:
Just to add more context. `aten.copy` and `aten.copy_` ops are not supported by reference TS LTC backend, so they doesn't work out-of-the box. But as they have CPU fallback they successfully compute these ops which is not the case for out backend. In order to support full codegen for these ops we need this extra shape inference function `compute_shape_copy` ([which is missing in PyTorch](https://github.com/pytorch/pytorch/blob/master/torch/csrc/lazy/core/shape_inference.h)). Ideally all these shape inference functions from `python/torch_mlir/csrc/base_lazy_backend/shape_inference.cpp` should be upstreamed to PyTorch, but as LTC flow is still under development (for example, in XLA LTC backend they [use partial codegen](https://github.com/pytorch/xla/blob/f763d0d2a4dac0872e2e1670332fd490f524b8be/xla_native_functions.yaml#L162-L163) for `aten.copy` ops and  manually lower them https://github.com/pytorch/xla/blob/f763d0d2a4dac0872e2e1670332fd490f524b8be/torch_xla/csrc/tensor_methods.cpp#L2172-L2180) so we decided to wait sometime until things stabilize.

GlebKazantaev commented:
@rahuls-cerebras it seems like we fixed some of the tests:
<img width="485" alt="image" src="https://user-images.githubusercontent.com/18436845/231319703-69e36902-ae2e-4ac2-b364-d03095db38f8.png">
could you please update them?

rahuls-cerebras commented:
> @rahuls-cerebras it seems like we fixed some of the tests: <img alt="image" width="485" src="https://user-images.githubusercontent.com/18436845/231319703-69e36902-ae2e-4ac2-b364-d03095db38f8.png"> could you please update them?

fixed

rahuls-cerebras commented:
@ramiro050 can you please give me appropriate permission to merge the PR

ramiro050 commented:
> rahuls-cerebras

Done! You should've gotten an email
